### PR TITLE
DEVTOOLS: Replace ugly SDL/MinGW hack with a cleaner hack

### PR DESCRIPTION
**Pull request by orgads**
Instead of having `main=SDL_main` defined by `pkg-config`/`sdl2-config --cflags`
and undefining it everywhere, strip it in its origin.

### CODE REVIEW

**Review by lotharsm — APPROVED** [review]
Looking good to me - thanks!

File: configure [hunk 1]
```diff
@@ -3790,7 +3790,7 @@ if test "$_sdlnet" = auto ; then
 	if test "$_sdl" = no || test "$_sdlMajorVersionNumber" = 2; then
 		if test "$_pkg_config" = "yes" && $_pkgconfig --exists SDL2_net; then
 			append_var SDL2_NET_LIBS "`$_pkgconfig --libs SDL2_net`"
-			append_var SDL2_NET_CFLAGS "`$_pkgconfig --cflags SDL2_net`"
+			append_var SDL2_NET_CFLAGS "`$_pkgconfig --cflags SDL2_net | sed 's/[[:space:]]*-Dmain=SDL_main//g'`"
```
Comment (lephilousophe) [inline configure:3793]: Is it really needed here?

File: configure [hunk 2]
```diff
@@ -3799,7 +3799,7 @@ if test "$_sdlnet" = auto ; then
 	if test "$_sdl" = no || test "$_sdlMajorVersionNumber" = 1; then
 		if test "$_pkg_config" = "yes" && $_pkgconfig --exists SDL_net; then
 			append_var SDL1_NET_LIBS "`$_pkgconfig --libs SDL_net`"
-			append_var SDL1_NET_CFLAGS "`$_pkgconfig --cflags SDL_net`"
+			append_var SDL1_NET_CFLAGS "`$_pkgconfig --cflags SDL_net | sed 's/[[:space:]]*-Dmain=SDL_main//g'`"
```
Comment (lephilousophe) [inline configure:3802]: and here?

**Comment by lephilousophe** [comment]
Overall, I don't understand the idea to request review and not even wait for a day before merging.
I agree PR must not get forgotten but taking some days is not a problem at all.

File: configure (hunk 1 above)
Comment (orgads) [reply configure:3793]: Yes, because SDL_net "requires" sdl2, so it returns its flags too.

File: configure (hunk 2 above)
Comment (orgads) [reply configure:3802]: Not sure about this one, but it shouldn't hurt.

**Comment by orgads** [comment]
Trust me, I'm an engineer. 😉

Just kidding. I'll wait longer next time. One thing I like about gerrit is that a reviewer can vote +1, still requiring another reviewer to vote +2 before merging. A feature I miss on github.

**Pull request merged**